File: Maps/CanPair/pair.py
class Solution:
    def canPair(self, arr, k):
        #code here.
        n = len(arr)
        if n%2!=0: return False
        mp = {}
        for i in range(n):
            rem = arr[i]%k
            mp[rem] = mp.get(rem,0) + 1
        for ele in mp:
            target = k-ele
            if ele == target or ele == 0:
                if mp[ele]%2!=0:
                    return False
            else: 
                if mp.get(target,0): 
                    if mp[ele]!=mp[target]:
                        return False
                else: return False
        return True
            

File: Maps/CanPair/pair.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
  public:
    bool canPair(vector<int> &arr, int k) {
        int n = arr.size();
        if (n%2!=0) return false;
        unordered_map<int,int>mp;
        for(int i=0;i<n;i++){
            int rem = arr[i]%k;
            mp[rem]++;
        }
        for(auto p: mp){
            int ele = p.first;
            int target = k-ele;
            if(ele == target || ele == 0){
                if (mp[ele]%2!=0)
                    return false;
            }
            else{
                if(mp[target]){
                    if (mp[ele]!=mp[target]) return false;
                }
                else return false;
            }
        }
        return true;
    }
};